fenicsprecice/adapter_core.py

```diff
@@ -386,19 +389,59 @@ def edge_is_on(subdomain, this_edge):
 
     vertices1_ids = []
     vertices2_ids = []
+    edges_ids = []
 
     for edge in edges(function_space.mesh()):
         if edge_is_on(coupling_subdomain, edge):
             v1, v2 = list(vertices(edge))
             if v1.global_index() in global_ids and v2.global_index() in global_ids:
                 vertices1_ids.append(id_mapping[v1.global_index()])
                 vertices2_ids.append(id_mapping[v2.global_index()])
+                edges_ids.append(edge.index())
 
     vertices1_ids = np.array(vertices1_ids)
     vertices2_ids = np.array(vertices2_ids)
+    edges_ids = np.array(edges_ids)
+
+    return vertices1_ids, vertices2_ids, edges_ids
+
+def get_coupling_triangles(function_space, coupling_subdomain, global_ids, precice_edge_dict):
+    """
+    Extracts triangles of mesh which lie on the coupling region.
+
+    Parameters
+    ----------
+    function_space : FEniCS function space
+        Function space on which the finite element problem definition lives.
+    coupling_subdomain : FEniCS Domain
+        FEniCS domain of the coupling interface region.
+    global_ids: numpy_array
+        Array of global IDs of vertices owned by this rank.
+
+
+    Returns
+    -------
+    edges : numpy array
+        Array of edges indices (3 per triangle)
+
+    """
+
+    def cell_is_in(subdomain, this_cell):
+        """
+        Check whether edge lies within subdomain
+        """
+        assert(len(list(vertices(this_cell))) == 3), "Only triangular meshes are supported"
+        return all([subdomain.inside(v.point(), True) for v in vertices(this_cell)])
+
+    edges_ids = []
 
-    return vertices1_ids, vertices2_ids
+    for cell in cells(function_space.mesh()):
+        if cell_is_in(coupling_subdomain, cell):
+            e1, e2, e3 = list(edges(cell))
+            if e1.index() in precice_edge_dict.keys() and e2.index() in precice_edge_dict.keys() and e3.index() in precice_edge_dict.keys():
```

Comment:
Where is the variable `precice_edge_dict` defined? Is it somehow a global variable in this file?